addons/filesystem/functions/fnc_device_addFile.sqf

```diff
@@ -0,0 +1,114 @@
+/**
+ * PUBLIC
+ *
+ * Adds a file to a given Device. Device could be a comupter or a flash drive.
+ *
+ * Arguments:
+ * 1: Device <OBJECT>
+ * 2: Path <STRING>
+ * 3: Content <STRING>
+ * 4: isCode <BOOL>
+ * 5: Owner <STRING>
+ * 6: Permissions <[ARRAY]>
+ * 7: isEncrypted <BOOL> Optional
+ * 8: encryptionAlgorithm <STRING> Optional
+ * 9: encryptionKey <STRING> Optional
+ *
+ * Results:
+ * Logs if file already exists; On other errors it throws an exception
+ *
+ * Examples:
+ * [_device, "/tmp/new/example.txt", "Lorem ipsum dolor sit amet", false, "root", [[false, true, true], [false, true, true]]] call AE3_filesystem_fnc_device_addFile;
+ * [_device, "/test.prg", "hint 'hello world';", true, "root", [[true, false, false], [true, false, false]]] call AE3_filesystem_fnc_device_addFile;
+ * [_device, "/tmp/password.txt", "secret", false, "root", [[false, true, true], [false, true, true]], true, "caesar", "13"] call AE3_filesystem_fnc_device_addFile;
+ *
+ * Permissions:
+ * [[owner execute, owner read, owner write], [everyone execute, everyone read, everyone write]]
+ */
+
+params ["_computer", "_path", "_content", "_isCode", "_owner", "_permissions", ["_isEncrypted", false], ["_encryptionAlgorithm", nil], ["_encryptionKey", nil]];
+
+private _filesystem = _computer getVariable "AE3_filesystem";
+
+if(_isCode) then
+{
+	_content = compile _content;
+};
+
+if (_isEncrypted) then 
+{
+	private _mode = "encrypt";
```

Comment:
Should be its own function. But I remember, that we already argued over this bit.

Reply:
Yes, that's true. If this code block grows further in the future, we will source out this code block to it's own function.